### PR TITLE
Add CurrencySnapshotEvents parameter to Rewards.distribute

### PR DESCRIPTION
**Pull request by sdavidp**

### CODE REVIEW

File: modules/sdk/src/main/scala/org/tessellation/sdk/domain/rewards/Rewards.scala [hunk 1]
```diff
@@ -7,13 +7,15 @@ import org.tessellation.schema.balance.Balance
...
 import org.tessellation.security.signature.Signed
 
 trait Rewards[F[_], P <: StateProof, S <: IncrementalSnapshot[P]] {
   def distribute(
     lastArtifact: Signed[S],
     lastBalances: SortedMap[Address, Balance],
     acceptedTransactions: SortedSet[Signed[Transaction]],
-    trigger: ConsensusTrigger
+    trigger: ConsensusTrigger,
+    events: Set[CurrencySnapshotEvent]
```
Comment (marcinwadon) [inline modules/sdk/src/main/scala/org/tessellation/sdk/domain/rewards/Rewards.scala:19]: looks like it leaks the implementation to the declaration. at this point we know nothing about `S` being a global snapshot or currency snapshot. I think you should parametrize `events` and pass global snapshot events on GL0 and currency snapshot events on Currency L0

**Review by marcinwadon — CHANGES_REQUESTED** [review]

File: modules/sdk/src/main/scala/org/tessellation/sdk/domain/rewards/Rewards.scala (hunk 1 above)
Comment (sdavidp) [reply modules/sdk/src/main/scala/org/tessellation/sdk/domain/rewards/Rewards.scala:19]: @ryle-goehausen ☝️ FYI

File: modules/core/src/main/scala/org/tessellation/infrastructure/snapshot/GlobalSnapshotConsensusFunctions.scala [hunk 2]
```diff
@@ -120,7 +120,7 @@ object GlobalSnapshotConsensusFunctions {
             snapshotContext,
             lastActiveTips,
             lastDeprecatedTips,
-            rewards.distribute(lastArtifact, snapshotContext.balances, _, trigger),
+            rewards.distribute(lastArtifact, snapshotContext.balances, _, trigger, Set.empty),
```
Comment (marcinwadon) [inline modules/core/src/main/scala/org/tessellation/infrastructure/snapshot/GlobalSnapshotConsensusFunctions.scala:123]: imo you should pass `events` here, not the `Set.empty`

**Review by marcinwadon — CHANGES_REQUESTED** [review]

File: modules/core/src/main/scala/org/tessellation/infrastructure/snapshot/GlobalSnapshotConsensusFunctions.scala (hunk 2 above)
Comment (sdavidp) [reply modules/core/src/main/scala/org/tessellation/infrastructure/snapshot/GlobalSnapshotConsensusFunctions.scala:123]: I think @ryle-goehausen asked me to pass empty set...

Comment (marcinwadon) [reply modules/core/src/main/scala/org/tessellation/infrastructure/snapshot/GlobalSnapshotConsensusFunctions.scala:123]: isn't it misleading that I expect it to be a set of events and I got an empty list? seems to be a bug 🤔

Comment (ryle-goehausen) [reply modules/core/src/main/scala/org/tessellation/infrastructure/snapshot/GlobalSnapshotConsensusFunctions.scala:123]: Right now the only purpose of this event set is for use in an application rewards function -- we're not calculating any rewards from any other events yet -- if in the future we do so we can pass something here.

Comment (ryle-goehausen) [reply modules/core/src/main/scala/org/tessellation/infrastructure/snapshot/GlobalSnapshotConsensusFunctions.scala:123]: If this events is easily in scope here however, there's no problem passing it so long as it's not actually affecting any code. I was fine with empty set when we were talking about the other object typed differently. Here now after updating the types its fine to pass in (so long as we aren't using it.)

Comment (sdavidp) [reply modules/core/src/main/scala/org/tessellation/infrastructure/snapshot/GlobalSnapshotConsensusFunctions.scala]: Done

**Review by ryle-goehausen — APPROVED** [review]
requested changes implemented.

**Review by marcinwadon — APPROVED** [review]

**Pull request merged**